apis/core/v1alpha2/core_types.go

```diff
@@ -30,11 +30,24 @@ type DefinitionReference struct {
 	Name string `json:"name"`
 }
 
+type ChildResourceKind struct {
+	// ApiVersion of the child resource
+	ApiVersion string `json:"apiVersion"`
+
+	// Kind of the child resource
+	Kind string `json:"kind"`
+
+	// LabelSelector to select the resources that the workload wants to expose to traits
+	LabelSelector map[string]string `json:"labelSelector,omitempty"`
+}
+
 // A WorkloadDefinitionSpec defines the desired state of a WorkloadDefinition.
 type WorkloadDefinitionSpec struct {
-	// Reference to the CustomResourceDefinition that defines this workload
-	// kind.
+	// Reference to the CustomResourceDefinition that defines this workload kind.
 	Reference DefinitionReference `json:"definitionRef"`
+
+	// ChildResourceKinds are the list of GVK of the child resources this workload generates
+	ChildResourceKinds []ChildResourceKind `json:"childResourceKinds,omitempty"`
```

Comment:
I believe we were thinking of making these object references (https://github.com/crossplane/crossplane/pull/1454#discussion_r421949685)

Reply:
yeah, use object references is fine

Reply:
I am not sure how can they be object references. They are not objects, they are kind with selectors